operators/src/processing/reprojection.rs

```diff
@@ -0,0 +1,399 @@
+use std::marker::PhantomData;
+
+use futures::TryStreamExt;
+use geoengine_datatypes::{
+    operations::reproject::{CoordinateProjection, CoordinateProjector, Reproject},
+    spatial_reference::SpatialReference,
+};
+use serde::{Deserialize, Serialize};
+use snafu::ensure;
+
+use crate::{
+    engine::{
+        ExecutionContext, InitializedOperator, InitializedOperatorImpl, InitializedVectorOperator,
+        Operator, QueryContext, QueryRectangle, TypedVectorQueryProcessor, VectorOperator,
+        VectorQueryProcessor, VectorResultDescriptor,
+    },
+    error,
+    util::Result,
+};
+
+use super::map_query::MapQueryProcessor;
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct ReprojectionParams {
+    pub target_spatial_reference: SpatialReference,
+}
+
+pub type Reprojection = Operator<ReprojectionParams>;
+pub type InitializedReprojection =
+    InitializedOperatorImpl<(), VectorResultDescriptor, (SpatialReference, SpatialReference)>;
+
+#[typetag::serde]
+impl VectorOperator for Reprojection {
+    fn initialize(
+        self: Box<Self>,
+        context: &dyn ExecutionContext,
+    ) -> Result<Box<InitializedVectorOperator>> {
+        ensure!(
+            self.vector_sources.len() == 1,
+            error::InvalidNumberOfVectorInputs {
+                expected: 1..2,
+                found: self.vector_sources.len()
+            }
+        );
+        ensure!(
+            self.raster_sources.is_empty(),
+            error::InvalidNumberOfRasterInputs {
+                expected: 0..1,
+                found: self.raster_sources.len()
+            }
+        );
```

Comment:
maybe add a TODO here, that it supports rasters in the future.

Reply:
i guess we will add this as a raster operator

Reply:
Can't the reprojection operator just implement both?

Reply:
i guess we will see